src/rai/rai/tools/ros/debugging.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (C) 2024 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from subprocess import PIPE, Popen
+from threading import Timer
+from typing import List, Literal, Optional
+
+from langchain_core.tools import tool
+
+FORBIDDEN_CHARACTERS = ["&", ";", "|", "&&", "||", "(", ")", "<", ">", ">>", "<<"]
+
+
+def run_with_timeout(cmd: List[str], timeout_sec: int):
+    proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    timer = Timer(timeout_sec, proc.kill)
+    try:
+        timer.start()
+        stdout, stderr = proc.communicate()
+        return stdout, stderr
+    finally:
+        timer.cancel()
+
+
+def run_command(cmd: List[str], timeout: int = 5):
+    # Validate command safety by checking for shell operators
+    # Block potentially dangerous characters
+    if any(char in " ".join(cmd) for char in FORBIDDEN_CHARACTERS):
+        raise ValueError(
+            "Command is not safe to run. The command contains forbidden characters."
+        )
+    stdout, stderr = run_with_timeout(cmd, timeout)
+    output = {}
+    if stdout:
+        output["stdout"] = stdout.decode("utf-8")
+    else:
+        output["stdout"] = "Command returned no stdout output"
+    if stderr:
+        output["stderr"] = stderr.decode("utf-8")
+    else:
+        output["stderr"] = "Command returned no stderr output"
+    return str(output)
+
+
+@tool
+def ros2_action(
+    command: Literal["info", "list", "type"],
```

Comment:
```suggestion
    command: Literal["info", "list", "type", "send_goal"],
```

I tested it with turtlebot and for simple mission / ros stack debugging to worked great.  See pdf:
[ros2_action_spin.pdf](https://github.com/user-attachments/files/18007000/ros2_action_spin.pdf)